src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,30 @@
+import styled from '@emotion/styled';
+import { ReactNode } from 'react';
+
+// adding <BadgeProps> to tell it that the props are the same from Badge (ie. emphasis, text, etc)
+const StyledBadge = styled.div<BadgeProps>`
+  display: inline-block;
+  padding: 5px;
+  border-radius: 8px;
+  background: #D4EFDF;
+  ${({
+    // add in styling
+
+  }) =>
+  `
+  
+  `}
+`;
+interface BadgeProps {
+  children: ReactNode
+  // adding a ? before the colon makes it optional
+  emphasis?: 'default' | 'strong';
+}
```

Comment:
In typescript, there has been an on going debate in regards to interface vs type. I think it's time that you choose your side? 😄 

I suggest that you look up this guy in youtube. He is a wizard. He's name is Matt Pocock. 

For me, I only use interface when I extend a type or when if I am working in a typescript based API.

There really isn't a right or wrong answer here. 

For me, I normally use type when I am working in the frontend. I would mostly use interface when I am at the backend or when I need a type a bit more complicated. 

But everything you can do with a type, you can also do with interface. For example, the below does the same thing:

interface ButtonProps extends ComponentPropsWithoutRef<'button'> {
 variant:  'primary' | 'secondary'
 };
 
 type ButtonProps = ComponentPropsWithoutRef<'button'> & {
  variant: 'primary' | 'secondar'
  };

Again, there is no right or wrong answer. But whatever you choose, choose wisely! 😄 

Always check the what is the pattern used in the code base you're working in.

Reply:
@ctrl-alt-delete007 The rule I've been following, and suggesting to others, is to use `interface` until you need to use `type`. 

I believe I picked up that convention from here: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example#tldr

Reply:
And whoops, I should have explicitly said I don't care which way either. I more meant to say "it's my fault they did that way", and also  to add to the conversation with "here's where I got that idea from [link]".

Reply:
no worries, i didn't think that was any issue at all 😄 my intention was to just bring awareness on one of the most widely discussed topic in typescript. Pls feel free to advise our aspiring engineers as how you see fit. 😃 

I think this is awesome because they are seeing both sides here and both of us are providing resources to discover for themselves. Cheers!

Reply:
everything looks good - going to approve